rclpy/rclpy/utilities.py

```diff
@@ -38,3 +39,29 @@ def try_shutdown():
 
 def get_rmw_implementation_identifier():
     return _rclpy.rclpy_get_rmw_implementation_identifier()
+
+
+def timeout_sec_to_nsec(timeout_sec):
+    """
+    Convert timeout in seconds to rcl compatible timeout in nanoseconds.
+
+    Python tends to use floating point numbers in seconds for timeouts. This utility converts a
+    python-style timeout to an integer in nanoseconds that can be used by rcl_wait.
+
+    :param timeout_sec: Seconds to wait. Block forever if None or negative. Don't wait if 0
+    :type timeout_sec: float or None
+    :rtype: int
+    :returns: rcl_wait compatible timeout in nanoseconds
+    """
+    if timeout_sec is None:
+        # Block forever
+        return -1
+    elif timeout_sec > -1.0 / S_TO_NS and timeout_sec < 1.0 / S_TO_NS:
```

Comment:
The goal was only to avoid == on a floating point number. The bound can be smaller (gtest `EXPECT_DOUBLE_EQ` uses 4 ULP). I chose 1ns because it's the smallest amount of time `rcl_wait` supports. How about changing it to 1 ULP on the negative side (block forever if timeout_sec <  0), and 0.5ns on the positive side (don't wait if 0 == round(timeout_sec / S_TO_NS)) ?

Reply:
Not sure what I was thinking before, but in 8b2008c027042b719585ef95bdddd5fd9fe0a50b any negative number blocks, and a timeout less than 1ns doesn't block due to the integer cast.

Reply:
ok, it was to get the tests to pass. 8b2008c matches what I'd expect now